libraries/SPI/SPI.cpp

```diff
@@ -74,13 +75,17 @@ void SPIClass::setHwCs(bool use) {
 }
 
 void SPIClass::beginTransaction(SPISettings settings) {
-    while(SPI1CMD & SPIBUSY) {}
+    while(inTransaction || SPI1CMD & SPIBUSY) {
+        yield();
```

Comment:
I think this behaviour should be handled in user code. Some people might want GPIO interrupts during SPI transactions, if GPIO interrupts serve other purposes than starting SPI transactions.